db unit tests update

olgamelnichuk opened the pull request:
Updating db tests to use junit 4.x

Commit:
Preparing for hotfix releases, if any

Commit:
Merge branch 'master' into atlas-2.0.9-fixes

rpetry committed:
Modified to add not just .nc and asset files to ncdf tar ball, but the whole content of each experiment's ncdf directory

rpetry committed:
file no longer needed for packaging ncdfs

rpetry committed:
Modified to add not just .nc and asset files to ncdf tar ball, but the whole content of each experiment's ncdf directory - further mods to add each directory only once (and not as many times as there .nc files in that experiment)

alf239 committed:
Merge pull request #105 from gxa/rpetry

ncdf tar ball creation changes

Commit:
Trac 3092: package-ncdfs.sh creates recursive symlink in ncdf

Replaced the symlink trick with tar's -C option, cheers to @geometer for pointing at it

rpetry committed:
Merge pull request #106 from gxa/trac3092

Trac 3092: package-ncdfs.sh creates recursive symlink in ncdf

rpetry committed:
Ticket #3090 - changes to handle correctly gene identifiers containing plus and colon characters in url re-written gene/<geneid> and experiment/<expacc>/<geneid> type urls.

rpetry committed:
Ticket #3090 - Various code review comments and fixes

rpetry committed:
Ticket #3090 - Various code review comments and fixes

rpetry committed:
Ticket #3094 - Modified to not show new experiment count if it's equal to 0

olgamelnichuk committed:
Merge pull request #109 from gxa/trac3094

Ticket #3094 Do not show new experiments count if it's zero

rpetry committed:
Ticket #3084 - Fixed to observe min experiment constraint in list view

olgamelnichuk committed:
Merge pull request #110 from gxa/trac3084

Ticket #3084 Minimum experiments query condition (in advanced search interface) is not observed in results shown as list. Genes with up/down counters less then required minimum condition are shown in the list.

Commit:
Ticket #3091 Efvs with containing single quotes should be escaped when server-side javascript is being generated

rpetry committed:
Ticket #3090 - Replaced urlrewrite.xml rules for /gene/ and /experiment/ with Spring @PathVariable handling

Commit:
Ticket #3072 JSON/XML API string contains duplicated gene identifiers on expt page

Commit:
Ticket #2929 if "Autodetect" is not selected in the Tasks drop-down on Admin, checkboxes are ignored

Commit:
Ticket #3069 Invalid solr query "id:()" when search by existing gene but there is no experiments for it

rpetry committed:
Ticket #3080 - Fixed to resolved EFO terms correctly after all children ontology expansion in heatmap

rpetry committed:
Merge pull request #112 from gxa/trac3072

Ticket #3072 JSON/XML API string contains duplicated gene identifiers on expt page

rpetry committed:
Merge pull request #114 from gxa/trac3069

Ticket #3069 Invalid solr query "id:()" when search by existing gene but there is no experiments for it

rpetry committed:
Merge pull request #111 from gxa/trac3091

Ticket #3091 Efvs with containing single quotes should be escaped when server-side javascript is being generated

rpetry committed:
Ticket #3079 - A fix to exclude an upward-expanded ontology term in heatmap from the popped up tree of parents

rpetry committed:
Merge pull request #113 from gxa/trac2929

Ticket #2929 if "Autodetect" is not selected in the Tasks drop-down on Admin, checkboxes are ignored

rpetry committed:
Ticket #3080 - a small js syntax change

olgamelnichuk committed:
Merge pull request #115 from gxa/trac3080

Trac3080 - Fixed to resolve EFO terms correctly after all children ontology expansion in heatmap

rpetry committed:
Ticket #3090 - Re-instated @RequestParam-based Controller methods

olgamelnichuk committed:
Merge pull request #108 from gxa/trac3090

Ticket #3090 - changes to handle correctly gene identifiers containing plus and colon characters in url re-written gene/ and experiment// type urls.

olgamelnichuk committed:
Merge pull request #116 from gxa/trac3079

Ticket #3079 Upward expansion of ontology in heatmap should not offer "self" as a choice

rpetry committed:
Ticket #3090 - Fixed Solr escaping for finding genes  by name

olgamelnichuk committed:
Merge pull request #117 from gxa/trac3090

Ticket #3090 - Fixed Solr escaping for finding genes by name

rpetry committed:
Ticket #3090 - Added a missing @RequestMapping

Commit:
Ticket #3097 List of experiment plots not consistent on gene page
(efv list on gene page is case sensitive now)

Commit:
Ticket #3098 Link "show this factor only>>" is broken after moving urlRewriter rules to spring controller

rpetry committed:
Merge pull request #118 from gxa/trac3097

Ticket 3097 List of experiment plots not consistent on gene page

rpetry committed:
Merge pull request #119 from gxa/trac3098

Ticket #3098 Link "show this factor only>>" is broken after moving urlRewriter rules to spring controller

rpetry committed:
[maven-release-plugin] prepare for next development iteration

rpetry committed:
[maven-release-plugin] prepare release atlas-2.0.9.1

Commit:
Merge commit 'atlas-2.0.9.1'

nsklyar committed:
added GeneSigDB annotations

rpetry committed:
Ticket #3112 - Direct single gene queries to heatmap instead of the gene page

rpetry committed:
Ticket #3112 - Rollback of an incorrect change

nsklyar committed:
fixed sql script

rpetry committed:
Ticket #3112 - handle correctly gxa/gene/IDENTIFIER and gxa/gene/IDENTIFIER/EF queries where IDENTIFIER matches more than one gene

rpetry committed:
Ticket #3112 - Re-instated missing method parameter in javadoc

alf239 committed:
Merge pull request #134 from gxa/trac3112

Trac 3112 - Direct gene queries with multiple results to heatmap instead of the gene page

Commit:
A bit more explicit redirect definition

 - fixed and clear list of parameters
 - no parameter overrides

Commit:
Proper diff

rpetry committed:
Merge pull request #135 from gxa/trac3112

Following up trac 3112

nsklyar committed:
fixed sql scripts (added "EXIT" in the end)

rpetry committed:
Merge pull request #133 from gxa/add_genesigdb

Added annotations for GeneSigDB

rpetry committed:
Ticket #3113 - Introduced config-based way of displaying gene identifier-based links to external resource on gene page (case in point: GeneSigDB)

rpetry committed:
Ticket #3113 - <c:choose> tag doesn't seem to like comments inside it

rpetry committed:
Ticket #3103 - Cherry picked change from develop branch

rpetry committed:
Ticket #3108 - Cherry picked change from develop branch

rpetry committed:
Cherry picked from develop branch: f428e734e0f530599876b17cca63d1c9df87014f Added more curated names and external links for genes.

rpetry committed:
Cherry picked from develop branch: f428e734e0f530599876b17cca63d1c9df87014f Added more curated names and external links for genes. Addressed review comments.

ostolop committed:
Merge pull request #136 from gxa/trac3113

Trac3113

rpetry committed:
[maven-release-plugin] prepare release atlas-2.0.9.2

rpetry committed:
[maven-release-plugin] prepare for next development iteration

alf239 committed:
Merge commit 'atlas-2.0.9.2'

Commit:
Trac ticket 3119: `org.hibernate.AssertionFailure: no collection snapshot for orphan delete`

 - `Property` must be mutable
 - made it mutable, adjusted caching settings

http://bar.ebi.ac.uk:8080/trac/ticket/3119

geometer committed:
Merge pull request #140 from gxa/trac3119

Trac 3119

Commit:
Trac 3123: can not release data for experiment:Illegal attempt to associate a collection with two open sessions; nested exception is org.hibernate.HibernateException: Illegal attempt to associate a collection with two open sessions

rpetry committed:
Ticket  #3122 - include all (rather than just scoring) exeriments in EB-eye dump

rpetry committed:
Ticket  #3122 - throw unexpected exception when bit index contains experiments that are not in DB

rpetry committed:
Ticket  #3122 - minor bug fix

Commit:
Following up Trac 3123: replaced start/end session with @Transactional where reasonably possible.

Tests are or will be addressed in `develop` branch

For details on the default-constructor tricks, see https://jira.springsource.org/browse/SPR-3150

rpetry committed:
Ticket  #3122 - changed time (from 10 mins to 1hr) before checking again if EB-eye dump file has been generated successfully. Note that typical dump generation time is around 2 hrs. Changed notification email address.

rpetry committed:
Merge pull request #141 from gxa/trac3123

Trac 3123: Cannot change private status

alf239 committed:
Dropped (unused) pre-fetching

alf239 committed:
Ensure pre-fetching

rpetry committed:
Ticket  #3121 - Modified to retrieve 'deep' ArrayDesign (ie. incl de and gene mappings) so that de ids loaded with the experiment can be validated - thus avoiding 'no gene/de mappings found warnings' on experiment load. Removed superfluous array design validation method.

rpetry committed:
Ticket  #3121 - Removed unused method

alf239 committed:
Merge pull request #143 from gxa/trac3122

Trac 3122

alf239 committed:
Merge pull request #144 from gxa/trac3121

Trac3121

rpetry committed:
[maven-release-plugin] prepare release atlas-2.0.9.3

Commit:
Merge commit 'atlas-2.0.9.3'

alf239 committed:
Merge commit 'atlas-2.0.10'

Conflicts:
	atlas-analytics/pom.xml
	atlas-analytics/src/main/java/uk/ac/ebi/gxa/analytics/generator/service/ExperimentAnalyticsGeneratorService.java
	atlas-dao/pom.xml
	atlas-dao/src/main/java/uk/ac/ebi/gxa/dao/AtlasDAO.java
	atlas-dao/src/main/java/uk/ac/ebi/gxa/dao/BioEntityDAO.java
	atlas-dao/src/main/java/uk/ac/ebi/gxa/dao/SoftwareDAO.java
	atlas-data-storage/pom.xml
	atlas-index-api/pom.xml
	atlas-loader/pom.xml
	atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/dao/LoaderDAO.java
	atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/service/AtlasMAGETABLoader.java
	atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/service/AtlasNetCDFUpdaterService.java
	atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/service/ExperimentEditorService.java
	atlas-loader/src/main/resources/uk/ac/ebi/gxa/loader/loaderContext.xml
	atlas-model/pom.xml
	atlas-model/src/main/java/uk/ac/ebi/microarray/atlas/model/Property.java
	atlas-test/pom.xml
	atlas-utils/pom.xml
	atlas-web/pom.xml
	atlas-web/src/main/java/ae3/model/AtlasGene.java
	atlas-web/src/main/java/uk/ac/ebi/gxa/web/controller/GeneViewController.java
	atlas-web/src/main/resources/atlas.properties
	atlas-web/src/main/webapp/WEB-INF/atlasApplicationContext.xml
	atlas-web/src/main/webapp/WEB-INF/jsp/genepage/experiment-list.jsp
	atlas-web/src/main/webapp/WEB-INF/jsp/genepage/gene.jsp
	atlas-web/src/main/webapp/admin.html
	indexbuilder/pom.xml
	indexbuilder/src/main/java/uk/ac/ebi/gxa/index/builder/service/ExperimentAtlasIndexBuilderService.java
	indexbuilder/src/main/java/uk/ac/ebi/gxa/index/builder/service/PropertiesIndexBuilderService.java
	pom.xml

alf239 committed:
<version>2.0.10.1-SNAPSHOT</version>

rpetry committed:
Removed superfluous time fom the notification email subject

rpetry committed:
Ticket #3228  - Modified to also include children in efo-only queries

rpetry committed:
Ticket #3233 - Included non-de experiments in API output

rpetry committed:
Merge pull request #199 from gxa/trac3228

Trac3228

rpetry committed:
Merge pull request #200 from gxa/trac3233

Trac3233

rpetry committed:
[maven-release-plugin] prepare release atlas-2.0.10.1

rpetry committed:
[maven-release-plugin] prepare for next development iteration

rpetry committed:
Ticket #3267 - Fixed an infinite loop bug for multi-page query results in which some pages have some genes excluded from the heatmap

rpetry committed:
Merge pull request #202 from gxa/trac3267

Trac3267 -  Data download from a page in which some genes are excluded from one of the result pages never comes back

rpetry committed:
[maven-release-plugin] prepare release atlas-2.0.10.2

rpetry committed:
[maven-release-plugin] prepare release atlas-2.0.11

rpetry committed:
[maven-release-plugin] prepare for next development iteration

rpetry committed:
Prevented an NPE in /gxa/das/s4/stylesheet requests

rpetry committed:
Prevented polluting our logs with ServletExceptions for invalid (deprecated) queries still often issued to this handler by web crawlers

rpetry committed:
Added USER_AGENT column to the script's output

rpetry committed:
Ticket #3332 - Added new species in RNA-seq experiments loaded for Atlas 2.0.11

rpetry committed:
Changed 2.0.12-SNAPSHOT to 2.0.11.1-SNAPSHOT - to prepare for a point release

rpetry committed:
Changed 2.0.12-SNAPSHOT to 2.0.11.1-SNAPSHOT - to prepare for a point release

rpetry committed:
[maven-release-plugin] prepare release atlas-2.0.11.1

Commit:
Merge commit 'atlas-2.0.10.1'

Commit:
Merge commit 'atlas-2.0.10.2'

Commit:
Merge commit 'atlas-2.0.11'

Conflicts:
	annotator/pom.xml
	atlas-analytics/pom.xml
	atlas-dao/pom.xml
	atlas-data-storage/pom.xml
	atlas-index-api/pom.xml
	atlas-loader/pom.xml
	atlas-model/pom.xml
	atlas-test/pom.xml
	atlas-updates/pom.xml
	atlas-utils/pom.xml
	atlas-web/pom.xml
	atlas-web/src/main/java/ae3/service/Download.java
	atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/result/HeatmapResultAdapter.java
	atlas-wro4j/pom.xml
	indexbuilder/pom.xml
	pom.xml

Commit:
Merge commit 'atlas-2.0.11.1'

Commit:
Reverted to proper atlas-2.0.11.1

rpetry committed:
New bitstats index for junit test

rpetry committed:
Fixed tests for the new bitstats test index

Commit:
db unit tests cleanup

Commit:
- surefire-plugin version updated to 2.11
- atlas-analytics: tests updated to run under junit 4.x

Commit:
closing sql connection after use

Commit:
- updated surefire-maven-plugin to version 2.10
- moved dbunit dependency version to the root pom

Commit:
removed unused code

Commit:
downgraded maven-surefire-plugin version to 2.9; as in 2.10 version there is a bug of detecting failed tests when run in a batch mode

alf239 commented:
That's cool. Could we please do two more things:
1. Check that _all_ test cases are bing run (I can only imagine inserting `fail()` in each, but that's time consuming
2. Update that nice ticket about 12 vs 1 assay? 

Additionally, this pull request gets us back to the red state (e.g., due to the ticket mentioned)—are we fine with that? Maybe it might make sense to fix the test (sorry cannot recall the ID) as well?

Commit:
postpone testing R service for a while...

Commit:
updated all biocep.properties with the new database connection params..

Commit:
added surefire junit provider to force tests run with junit 4.x

Commit:
making tests runnable again...

Commit:
switched to surefire-junit4, as surefire-junit47 too advanced for us yet, as it eliminates usage of test class hierarchy..

olgamelnichuk committed:
trying surefire-maven-plugin 2.8.1...

Commit:
-passed java.library.path to the unit tests to fix R-based failed tests;
-surefire plugin v.2.8.x writes strange symbols like @SL@SL@SL.. in the log, so its version downgraded to 2.7.2;

Commit:
checking R_HOME environment variable on bamboo....

Commit:
bamboo test: ignoring R tests for a while..

Commit:
bamboo test: revive just one R test

Commit:
bamboo test: revive one more R test..

Commit:
removed unused private method

Commit:
fixing R factory validation tests..

Commit:
fixing R factory validation tests..

Commit:
fixing R factory validation tests..

Commit:
Ignoring LocalR factory building test

olgamelnichuk committed:
a few info lines in the log added..

olgamelnichuk committed:
enabling local R factory builder  test

olgamelnichuk committed:
updated path to the native R libraries

olgamelnichuk committed:
- rolled back last change as it doesn't help;
- experimenting with surefire 2.9

olgamelnichuk committed:
rolled back to surefire 2.7.2

olgamelnichuk committed:
disabled local R factory builder test as its unix environment is not properly configured on bamboo

olgamelnichuk committed:
enabled compute task test...

Commit:
added a bit of secret knowledge about how to run local R...

Commit:
experimenting with LD_LIBRARY_PATH...

Commit:
one more try to configure local R services on bamboo..

Commit:
optimized imports

Commit:
a few code decorations..

olgamelnichuk committed:
fixing bamboo logs maven-surefire-report-plugin v 2.9

Commit:
Merge remote branch 'origin/dbTestsFix' into dbTestsFix

Commit:
fixing bamboo unit test reports: surefire-report-plugin v2.7.2

Commit:
fixing bamboo unit test reports: surefire-report-plugin v2.9

olgamelnichuk committed:
- fixed TestExperimentAnalyticsGeneratorService: added assays and samples from E-GEOD-5053.nc
- reorganized atlas-be-db.xml a bit to easier read experiment data
- converted all uppercase attribute names to lowercase

alf239 commented on atlas-analytics/src/main/java/uk/ac/ebi/gxa/R/LocalAtlasRFactory.java:
```diff
@@ -34,57 +34,69 @@
+ *     + JRI library must be in the current directory or any directory listed in java.library.path. Alternatively
+ *       you can specify its path with -Djava.library.path= when starting the JVM. When you use the latter,
+ *       make sure you check java.library.path property first such that you won't break your Java.
+ *
```
Sorry for being a pain in the neck: could you please add HTML here? `<ul>` and `<li>` will be helpful; IDEA's autoformatting feature will put `<p/>` for you at no cost.

alf239 replied:
`-Djava.library.path=` can be nicely highlighted with e.g., `<code>`

olgamelnichuk replied:
aaah completely forgot about it.. my bad

alf239 commented on atlas-analytics/src/test/resources/biocep.properties:
```diff
@@ -21,9 +21,9 @@
+biocep.db.url=jdbc:oracle:thin:@burns.ebi.ac.uk:1521:DWEP
 biocep.db.user=DWEP
-biocep.db.password=DWEP
+biocep.db.password=dwep2244
```
I strongly believe we should avoid mentioning internal infrastructure in the public code. Mentioning database passwords is a red flag, please fix.

(I admit it's not your fault: still, it's something that MUST be fixed asap)

olgamelnichuk replied:
agreed...

alf239 commented on atlas-test/src/main/resources/biocep.properties:
```diff
@@ -21,9 +21,9 @@
-biocep.db.password=dwep
+biocep.db.url=jdbc:oracle:thin:@burns.ebi.ac.uk:1521:DWEP
+biocep.db.user=DWEP
+biocep.db.password=dwep2244
```
As before, a big no-no. Even if noone cares about security, it's such a bad practice that it MUST be avoided (due to NDA clause if you prefer the strong reasons over a good taste)

alf239 commented on atlas-web/src/test/resources/biocep.properties:
```diff
@@ -21,13 +21,13 @@
+biocep.db.url=jdbc:oracle:thin:@burns.ebi.ac.uk:1521:DWEP
 biocep.db.user=DWEP
-biocep.db.password=DWEP
+biocep.db.password=dwep2244
```
Please eliminate

alf239 commented on atlas-web/src/test/resources/biocep.properties:
```diff
@@ -21,13 +21,13 @@
 
 # database-specific biocep properties
-biocep.db.url=jdbc:oracle:thin:@apu.ebi.ac.uk:1521:AEDWT
+biocep.db.url=jdbc:oracle:thin:@burns.ebi.ac.uk:1521:DWEP
```
internal infrastructure details MUST be passed as environment variables and substituted on resource copy rather than provided in plain text on a public site.

olgamelnichuk committed:
optimized imports

olgamelnichuk committed:
- deleted unused biocep.properties from atlas-test and atlas-web->test
- db connection prop values are extracted from biocep.properties and moved to the maven settings.xml

olgamelnichuk committed:
class javadoc reformatted

Commit:
Merge branch 'dbTestsFix', remote branch 'origin' into dbTestsFix

Commit:
argLine fix

Commit:
Merge remote branch 'origin/dbTestsFix' into dbTestsFix

Commit:
biocep.properties moved to the main->resources;

Commit:
biocep.properties moved to the main->resources;

Commit:
biocep.properties moved to the main->resources;

Commit:
Merge remote branch 'origin/dbTestsFix' into dbTestsFix

Conflicts:
	atlas-analytics/pom.xml

alf239 commented:
Apparently, an erroneous merge.
